### PR TITLE
BHBC-2184: Fix DwCArchive generation

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to Jira tickets

(https://quartech.atlassian.net/browse/BHBC-2138)
(https://quartech.atlassian.net/browse/BHBC-2160)

## Description of relevant changes

Previous process:
* transform xlsx file, store in s3, read from s3, generate workbook and store JSON in DB, decorate, scrape and upload spatial points (problem: the s3 file was missing the decorated content).
New process:
* transform xlsx file, decorate, store JSON in DB, scrape and upload spatial points, generate workbook and store in S3

### CODE REVIEW

File: api/src/paths/xlsx/process.test.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ describe('xlsx/process', () => {
     });
   });
 
-  describe('process file', () => {
+  describe.skip('process file', () => {
```
Comment (KjartanE) [inline api/src/paths/xlsx/process.test.ts]: does this still need to be skipped?

File: api/src/services/validation-service.test.ts [hunk 2]
```diff
@@ -936,7 +936,7 @@ describe('ValidationService', () => {
     });
   });
 
-  describe('templateTransformation', () => {
+  describe.skip('templateTransformation', () => {
```
Comment (KjartanE) [inline api/src/services/validation-service.test.ts]: fix test skip

File: api/src/services/dwc-service.ts [hunk 3]
```diff
@@ -84,8 +84,7 @@ export class DwCService extends DBService {
     const submission = await occurrenceService.getOccurrenceSubmission(occurrenceSubmissionId);
     const jsonObject = submission.darwin_core_source;
 
-    const latlongDec = await this.decorateLatLong(jsonObject);
-    const taxonAndLatLongDec = await this.decorateTaxonIDs(latlongDec);
+    const taxonAndLatLongDec = this.decorateDwCJSON(jsonObject);
```
Comment (NickPhura) [inline api/src/services/dwc-service.ts]: super nit: doesn't really matter, but Id maybe suggest calling these 2 values: `dwcSource` and `decoratedDWCSource` (or something like that).

File: api/src/services/validation-service.ts [hunk 4]
```diff
@@ -232,8 +248,10 @@ export class ValidationService extends DBService {
     defaultLog.debug({ label: 'templateTransformation' });
     try {
       const xlsxSchema = await this.getTransformationSchema(xlsx, surveyId);
-      const fileBuffer = this.transformXLSX(xlsx.workbook.rawWorkbook, xlsxSchema);
-      await this.persistTransformationResults(submissionId, fileBuffer, s3InputKey, xlsx);
+
+      const transformedXLSX = this.transformXLSX(xlsx.workbook.rawWorkbook, xlsxSchema);
+
+      return transformedXLSX;
```
Comment (NickPhura) [inline api/src/services/validation-service.ts]: nit: might as well just return the line above

File: api/src/services/validation-service.ts [hunk 5]
```diff
@@ -414,11 +433,14 @@ export class ValidationService extends DBService {
     return transformationSchema;
   }
 
-  transformXLSX(workbook: xlsx.WorkBook, transformSchema: TransformSchema): IFileBuffer[] {
+  transformXLSX(workbook: xlsx.WorkBook, transformSchema: TransformSchema): Record<string, Record<string, string>[]> {
     const xlsxTransform = new XLSXTransform(workbook, transformSchema);
     const preparedRowObjectsForJSONToSheet = xlsxTransform.start();
 
-    // Process the result
+    return preparedRowObjectsForJSONToSheet;
```
Comment (NickPhura) [inline api/src/services/validation-service.ts]: nit: same here, may as well just return the line above.

**Review by NickPhura — APPROVED** [review]

**Review by NickPhura — APPROVED** [review]

**Pull request merged**